src/garage/envs/multi_task_env.py

```diff
@@ -0,0 +1,139 @@
+"""
+A wrapper env that handles multiple tasks from different envs.
+
+Useful while training multi-task reinforcement learning algorithms.
+"""
+
+import akro
+import numpy as np
+
+from garage.envs import GarageEnv
+
+
+class MultiTaskEnv(GarageEnv):
+    """
+    A wrapper class to handle multi-task environments.
+
+    Args:
+        task_env_cls_dict (dict(str, garage.envs.GarageEnv)):
+            A dictionary of task_name and corresponding GarageEnv pairs.
+        task_args (dict): Arguments to be passed to GarageEnv constructer.
+            Keys should be same as keys of task_env_cls_dict.
+        task_kwargs (dict): Arguments to be passed to GarageEnv constructer.
+            Keys should be same as keys of task_env_cls_dict.
+        sample_strategy (str): The sample strategy to be used when
+            "sample_tasks" is called. Acceptable values are 'all' and 'random'.
+    """
+
+    def __init__(self,
+                 task_env_cls_dict,
+                 task_args,
+                 task_kwargs,
+                 sample_strategy='all'):
+
+        self._task_envs = []
+        self._active_task = 0
+        self._valid_sample_strategies = ['all', 'random']
```

Comment:
I think there are more than all and random? For example, random can be uniform or under other distributions.

Reply:
I think Utkarsh already encoded here that the environments all need to have unified spaces.

I can understand wanting a feature (or wrapper) where we automatically merge and pad mismatched spaces, but I don't think we should do that automatically. It could lead to really hard to find bugs for the most common case (which is matching spaces).